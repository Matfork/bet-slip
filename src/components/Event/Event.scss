@import "src/assets/scss/spacing";
@import "src/assets/scss/colors";

.Event-component {
  .MuiCardHeader-root {
    background: $blue;
    color: $white;
    text-align: center;
  }
  .Event-component__card {
    margin-bottom: $spacing-4;

    .Event-component__label {
      display: block;
      text-align: center;
    }

    .Event-component__markets {
      display: block;
    }
  }
}
