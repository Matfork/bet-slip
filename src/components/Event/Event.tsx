import { Card, CardContent, CardHeader } from '@material-ui/core';
import React from 'react';
import { useSelector } from 'react-redux';
import { RootState } from 'store/reducers';
import { EventsInitialState } from 'store/reducers/events.reducer';
import MarketComponent from './components/Market/Market';
import './Event.scss';

const className = 'Event-component';

const Event: React.FC = () => {
  const { events } = useSelector<RootState, EventsInitialState>((state) => state.events);

  return (
    <div className={className}>
      {events
        .filter((event) => event.markets.length)
        .map((event) => (
          <Card variant="outlined" key={event.id} className={`${className}__card`}>
            <CardHeader title={event.name}>
              <div className={`${className}__label`}>{event.name}</div>
            </CardHeader>
            <CardContent>
              <div className={`${className}__markets`}>
                <MarketComponent markets={event.markets} />
              </div>
            </CardContent>
          </Card>
        ))}
    </div>
  );
};

export default Event;
