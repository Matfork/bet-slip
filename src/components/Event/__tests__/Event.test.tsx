import { screen } from '@testing-library/react';
import { initialStateStub } from 'test.stub';
import { customRender } from 'test.utilities';
import Event from '../Event';

describe('Event Component', () => {
  it('should test snapshot', () => {
    const component = customRender(<Event />);
    expect(component.container).toMatchSnapshot();
  });

  it('should display all events avaiable', () => {
    customRender(<Event />);
    const titleEl = screen.queryByText(/Real Madrid vs Barcelona/);
    expect(titleEl).toBeInTheDocument();
  });

  it('should not display events if there is no data', () => {
    customRender(<Event />, {
      initialState: {
        ...initialStateStub,
        events: { ...initialStateStub.events, events: [] },
      },
    });
    const titleEl = screen.queryByText(/Real Madrid vs Barcelona/);
    expect(titleEl).toBeNull();
  });
});
