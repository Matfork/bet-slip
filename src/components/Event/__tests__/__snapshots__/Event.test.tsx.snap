// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component should test snapshot 1`] = `
<div>
  <div
    class="Event-component"
  >
    <div
      class="MuiPaper-root MuiCard-root Event-component__card MuiPaper-outlined MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Real Madrid vs Barcelona
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      >
        <div
          class="Event-component__markets"
        >
          <div
            class="Market-component"
          >
            <div
              class="Market-component__container"
            >
              <div
                class="Market-component__label"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6"
                >
                  Team to Win
                </h6>
              </div>
              <div
                class="Market-component__options"
              >
                <div
                  class="Selection-component"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        Real Madrid
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        1.23
                      </p>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        Barcelona
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        2.05
                      </p>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <hr
                class="MuiDivider-root"
              />
            </div>
            <div
              class="Market-component__container"
            >
              <div
                class="Market-component__label"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6"
                >
                  Player to Score First
                </h6>
              </div>
              <div
                class="Market-component__options"
              >
                <div
                  class="Selection-component"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        Ronaldo
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        1.15
                      </p>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        Messi
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        1.3
                      </p>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        Bale
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        1.35
                      </p>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <hr
                class="MuiDivider-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiCard-root Event-component__card MuiPaper-outlined MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Atletico Madrid vs Malaga
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      >
        <div
          class="Event-component__markets"
        >
          <div
            class="Market-component"
          >
            <div
              class="Market-component__container"
            >
              <div
                class="Market-component__label"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6"
                >
                  Team to Win
                </h6>
              </div>
              <div
                class="Market-component__options"
              >
                <div
                  class="Selection-component"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        Atletico
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        1.4
                      </p>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        Malaga
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body2"
                      >
                        3.05
                      </p>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <hr
                class="MuiDivider-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
